### PR TITLE
Add tip to use negative variations of Capybara matchers 

### PR DESCRIPTION
**Pull request by Tetramputechture**
When asserting that a selector is not present, we should use the negative variations of the Capybara finder function. If we use the positive form, Capybara will always wait the default finder timeout time.

For instance, using `assert page.has_no_css('ul')` executes immediately when Capybara does not find the selector, while using `assert_not page.has_css('ul')` *always* waits the Capybara default timeout time, and is functionally the same as before. 

This PR adds a tip for this in the Testing section of the Ruby style guide.

### CODE REVIEW

File: style/ruby/README.md [hunk 1]
```diff
@@ -240,6 +240,21 @@ https://github.com/hanmoi-choi/reek-emacs
     assert response['message'].include?('must be a valid email') # on error: expect false to be truthy
     ```
 
+- Use the negative variations of Capybara matchers if you are testing that an element or selector is not present.
+  Using the positive form of these matchers will always make Capybara wait
```
Comment (wafflewitch) [inline style/ruby/README.md:291]: should this say "wait for" ?

File: style/ruby/README.md [hunk 2]
```diff
@@ -240,6 +240,21 @@ https://github.com/hanmoi-choi/reek-emacs
...
+
+    ```ruby
+
+    # good
+
+    assert page.has_no_content?('Submit') # immediately executes if 'Submit' is not found
+
+    # bad
+
+    assert_not page.has_content?('Submit') # always waits the default Capybara timeout time
```
Comment (wafflewitch) [inline style/ruby/README.md:302]: "waits for" ?

**Review by wafflewitch — DISMISSED** [review]

File: style/ruby/README.md (hunk 2 above)
Comment (Tetramputechture) [reply style/ruby/README.md:302]: ditto

File: style/ruby/README.md (hunk 1 above)
Comment (Tetramputechture) [reply style/ruby/README.md:291]: 'for' is optional

**Review by onesneakymofo — APPROVED** [review]

**Pull request merged**